test/notification/index.js

```diff
@@ -118,48 +118,48 @@ describe("Notification", function() {
 
 		context("priority is non-default", function() {
 			it("contains the apns-priority header", function() {
-				note.priority = 5;
+				note.payload.priority = 5;
 				expect(note.headers()).to.have.property("apns-priority", 5);
 			});
 		});
 
 		context("id is set", function() {
 			it("contains the apns-id header", function() {
-				note.id = "123e4567-e89b-12d3-a456-42665544000";
+				note.payload.id = "123e4567-e89b-12d3-a456-42665544000";
 
 				expect(note.headers()).to.have.property("apns-id", "123e4567-e89b-12d3-a456-42665544000");
 			});
 		});
 
 		context("expiry is non-zero", function() {
 			it("contains the apns-expiration header", function() {
-				note.expiry = 1000;
+				note.payload.expiry = 1000;
 
 				expect(note.headers()).to.have.property("apns-expiration", 1000);
 			});
 		});
 
 		context("topic is set", function() {
 			it("contains the apns-topic header", function() {
-				note.topic = "io.apn.node";
+				note.payload.topic = "io.apn.node";
 
 				expect(note.headers()).to.have.property("apns-topic", "io.apn.node");
 			});
 		});
 
 		context("collapseId is set", function () {
 			it("contains the apns-collapse-id header", function () {
-				note.collapseId = "io.apn.collapse";
+				note.payload.collapseId = "io.apn.collapse";
 
 				expect(note.headers()).to.have.property("apns-collapse-id", "io.apn.collapse");
 			});
 		});
 
 		context("pushType is set", function () {
 			it("contains the apns-push-type header", function () {
-				note.pushType = "alert";
+				note.payload.pushType = "background";
 
-				expect(note.headers()).to.have.property("apns-push-type", "alert");
+				expect(note.headers()).to.have.property("apns-push-type", "background");
```

Comment:
Looks like these tests should be written.

`const note = new Notification({ pushType: 'alert' });`

Calling note.pushType sets it directly instead of taking it from the payload.

Reply:
I'm a little unsure what you mean here.

Initializing the notification this way in the tests causes the modified tests to fail; pushType doesn't seem to get added to the payload object. The test returns this:

```
  2) Notification
       headers
         pushType is set
           contains the apns-push-type header:
     AssertionError: expected {} to have a property 'apns-push-type'
```

Reply:
Oh I think I see what you mean.

But at runtime the values in question are in the payload object... I'm not sure what you're looking for here. Can you explain?